biochemist-python/chapters/Working_with_Pandas.ipynb

```diff
@@ -0,0 +1,1172 @@
+       "2    2ZNK  X-RAY DIFFRACTION        1.80         Thrombin       31U   \n",
+       "0    3SHC  X-RAY DIFFRACTION        1.90   Human Thrombin       B01   \n",
+       "8    2C8Z  X-RAY DIFFRACTION        2.14         Thrombin       C2A   \n",
+       "9    2C8Y  X-RAY DIFFRACTION        2.20         Thrombin       C3M   \n",
+       "10   2C90  X-RAY DIFFRACTION        2.25         Thrombin       C1M   \n",
+       "7    1UVT  X-RAY DIFFRACTION        2.50  Bovine Thrombin       I48   \n",
+       "6    1UVU  X-RAY DIFFRACTION        2.80  Bovine Thrombin       DCH   \n",
+       "\n",
+       "                                          Ligand name  \n",
+       "4   D-phenylalanyl-N-[(1-methylpyridinium-2-yl)met...  \n",
+       "5   D-phenylalanyl-N-(pyridin-4-ylmethyl)-L-prolin...  \n",
+       "1   D-phenylalanyl-N-(pyridin-3-ylmethyl)-L-prolin...  \n",
+       "3   D-phenylalanyl-N-(pyridin-2-ylmethyl)-L-prolin...  \n",
+       "2    D-leucyl-N-(4-carbamimidoylbenzyl)-L-prolinamide  \n",
+       "0   D-phenylalanyl-N-[(4-chloropyridin-2-yl)methyl...  \n",
+       "8                       1-(3-CHLOROPHENYL)METHANAMINE  \n",
+       "9   N-[(2R,3S)-3-AMINO-2-HYDROXY-4-PHENYLBUTYL]NAP...  \n",
+       "10                    1-(4-CHLOROPHENYL)-1H-TETRAZOLE  \n",
+       "7   N-{3-METHYL-5-[2-(PYRIDIN-4-YLAMINO)-ETHOXY]-P...  \n",
+       "6   3-(7-DIAMINOMETHYL-NAPHTHALEN-2-YL)-PROPIONIC ...  "
+      ]
+     },
+     "execution_count": 42,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "thrombin_df.sort_values('Resolution', inplace = True)\n",
+    "thrombin_df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"exercise-this-is-a-title exercise\">\n",
+    "<p class=\"exercise-title\">Exercise</p>\n",
+    "    <p> The enzyme kinetics data from the introduction can be found in the file `enzyme_kinetics.csv` in the data directory. The only problem is that the substrate concentrations are out of order. This is how it looks in Excel:</p>\n",
+    "    \n",
+    "<img src=\"images/enzyme_kinetics.png\" alt=\"enzyme kinetics data\" width=\"306\" height=\"387\"> \n",
+    "        \n",
+    "Create a dataframe from that file and use pandas functions to place the data in ascending substrate concentrations, then print the velocity at a substrate concentration of 20 micromolar. As a biochemist, you probably want to plot and analyze these data to find the Km and Vmax. We'll get to that in a bit. The goal right now is to just have a little fun with pandas.\n",
+    "    \n",
+    "```{admonition} Solution\n",
+    ":class: dropdown\n",
+    "    import os\n",
+    "    import pandas\n",
+    "    kinetics_file = os.path.join('data', 'enzyme_kinetics.csv')\n",
+    "    kinetics_df = pd.read_csv(kinetics_file) # kinetics_file is a variable, not a string\n",
+    "    kinetics_df.head(0)  # to find out the column labels\n",
```

Comment:
You can also use

```
kinetics_df.columns
```

To get the column names.

Reply:
I had not used  the df.columns option before, so I listed it in the comments for the solution.